src/main/java/bio/terra/service/filedata/azure/AzureSynapseService.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.service.filedata.azure;
+
+import static bio.terra.service.filedata.azure.AzureSynapsePdao.getDataSourceName;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.model.AccessInfoModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.resourcemanagement.MetadataDataAccessUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AzureSynapseService {
```

Comment:
TDR folks, I would especially like eyes here. For the `SnapshotBuilderService` I wanted to reuse some of the azure querying code, which already seemed repeated in `DatasetService`. This was what I thought would make sense for that service name and interfaces, but I definitely want eyes here.

Reply:
IMO, This code is pretty specific to datasets. I think it's fine to pull it out into a separate method and reference it the SnapshotBuilderService, but I think it should still live in the DatasetService. 
A "synapse service" would break naming conventions in TDR. Anything directly interacting with Synapse should be a pdao or some sort of utility. In this method, we're gathering info about the dataset and then feeding into the SynapsePdao which fits how we use the dataset service. However, it could be good to pull that "metadataUrl" piece into a util or something similar.

Reply:
@snf2ye I couldn't use it in DatasetService because then it created a dependency loop. To avoid that, it needs to be pulled out into its own service. (DatasetService->DatasetJsonConversion->SnapshotBuilderService->DatasetService)

I wasn't sure what to name that service though. This was the best I could come up with. Do you have any other better ideas?